### PR TITLE
ocaml-notty: init at 0.1.1

### PR DESCRIPTION
**Pull request by sternenseemann**
###### Motivation for this change

re-introduce notty

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/notty/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    rev    = "v${version}";
+    sha256 = "0bw3bq8z2y1rhc20zn13s78sazywyzpg8nmyjch33p7ypxfglf01";
+  };
+
+  buildInputs = [ findlib ];
+  propagatedBuildInputs = [ result uucp uuseg uutf ] ++
+                          optional withLwt [ lwt ];
+
+  configureFlags = [ "--enable-unix" ] ++
+                   optional withLwt ["--enable-lwt"];
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/notty/default.nix]: Square brackets should not be there.

Also, `enable` is the default: you should explicitly say `disable-lwt` if it is not available.

File: pkgs/development/ocaml-modules/notty/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  configurePhase = "./configure --prefix $out $configureFlags";
+
+  meta = with stdenv.lib; {
+    homepage = https://github.com/pqwy/notty/tree/master;
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/notty/default.nix]: This URL is strange; note that you can use `inherit (src.meta) homepage;`.

File: pkgs/development/ocaml-modules/notty/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+  buildInputs = [ findlib ];
+  propagatedBuildInputs = [ result uucp uuseg uutf ] ++
+                          optional withLwt [ lwt ];
+
+  configureFlags = [ "--enable-unix" ] ++
+                   optional withLwt ["--enable-lwt"];
+  configurePhase = "./configure --prefix $out $configureFlags";
+
+  meta = with stdenv.lib; {
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/notty/default.nix]: This `with stdenv.lib;` is not needed (see line 5).

File: pkgs/development/ocaml-modules/notty/default.nix (hunk 1 above)
Comment (sternenseemann) [inline pkgs/development/ocaml-modules/notty/default.nix]: Correct now?

File: pkgs/development/ocaml-modules/notty/default.nix (hunk 3 above)
Comment (sternenseemann) [inline pkgs/development/ocaml-modules/notty/default.nix]: Thanks, corrected it!

**Pull request merged**